content/docs/iac/using-pulumi/extending-pulumi/build-a-component.md

```diff
@@ -0,0 +1,2375 @@
+
+{{% /choosable %}}
+
+{{% choosable language go %}}
+
+First, create the `main.go` file, where we will define an entry point for the component.
+
+***Example:** `main.go` component entry point*
+
+```go
+package main
+
+import (
+	"github.com/pulumi/pulumi-go-provider/infer"
+)
+
+func main() {
+	err := infer.NewProviderBuilder().
+		WithName("static-page-component").
+		WithComponents(
+			infer.Component(NewStaticPage),
+		).
+		BuildAndRun()
+
+	if err != nil {
+		panic(err)
+	}
+}
+```
+
+Here, the `BuildAndRun` method creates and runs a Pulumi provider which acts as a shim for the component. We use `NewProviderBuilder()` to construct our provider and configure it with various parameters.
+
+The `WithName` method specifies the name of the component resource provider. This name will be important later on in the component implementation, so make sure it's something unique and descriptive!
+
+The `WithComponents` method registers the components that will be available through this provider. The `infer.Component` call tells the Pulumi SDK which types to look at for our component implementation. The `infer` library scans these structs and does a lot of heavy lifting to infer schema and necessary middleware, so that our component implementations can involve minimal coding.
+
+{{% /choosable %}}
+
+{{% choosable language csharp %}}
+
+First, create the `Program.cs` file, where we will define an entry point for the component.
+
+***Example:** `Program.cs` component entry point*
+
+```csharp
+using System.Threading.Tasks;
+
+class Program
+{
+    public static Task Main(string []args) =>
+        Pulumi.Experimental.Provider.ComponentProviderHost.Serve(args);
+}
+```
+
+Here, the `ComponentProviderHost.Serve` call invokes a Pulumi provider implmentation which acts as a shim for the component. Everything else about your component will be inferred by the Pulumi SDK.
+
+{{% /choosable %}}
+
+{{% choosable language java %}}
+
+First, create the `src/main/java/staticpagecomponent` sub-directory and in it, create the `App.java` file, where we will define an entry point for the component.
+
+***Example:** `App.java` component entry point*
+
+```java
+package staticpagecomponent;
+
+import java.io.IOException;
+import com.pulumi.provider.internal.Metadata;
+import com.pulumi.provider.internal.ComponentProviderHost;
+
+public class App {
+    public static void main(String[] args) throws IOException, InterruptedException {
+        new ComponentProviderHost(new Metadata("static-page-component"), App.class.getPackage()).start(args);
+    }
+}
+```
+
+Here, the `ComponentProviderHost.start(...)` call invokes a Pulumi provider implmentation which acts as a shim for the component. The `Metadata` we pass to it specifies the name of the component. This name will be important later on in the component implementation, so make sure it's something unique and descriptive!
```

Comment:
```suggestion
Here, the `ComponentProviderHost.start(...)` call invokes a Pulumi provider implmentation which acts as a shim for the component. The name we pass to it will be important later on in the component implementation, so make sure it's something unique and descriptive!
```